resources/docker_files/ubuntu-16.04/Dockerfile

```
@@ -117,7 +117,8 @@ RUN wget -q https://developer.arm.com/-/media/Files/downloads/compiler/DS500-PA-
 
 ENV ARMC5_BIN_DIR=/usr/local/ARM_Compiler_5.06u3/bin/
 ENV PATH=$PATH:/usr/local/ARM_Compiler_5.06u3/bin
-ENV ARMLMD_LICENSE_FILE=8225@armcc.licenses.mbedcloudtesting.com
+ARG ARMLMD_LICENSE_FILE=8225@armcc.licenses.mbedcloudtesting.com
```

This license server has been dead for a while, we just never got around to removing it.
Please replace the default with the internal CI's values, as per [Gilles's comment](https://github.com/ARMmbed/mbedtls-test/pull/34/commits/54305920b3cb055a5766ba2a83ca10f1aafca114#r799656032).

```suggestion
ARG ARMLMD_LICENSE_FILE=7010@10.6.26.52:7010@10.6.26.53:7010@10.6.26.54:7010@10.6.26.56
```

---

This change also applies to ubuntu-18.04/Dockfile, right?

---

Yes. Please note that [we're rewriting the 18.04 dockerfile to apply a number of changes we previously made to the 16.04 dockerfile](https://github.com/ARMmbed/mbedtls-test/pull/31). I think it would be easier to rebase your changes on top of #31 since #31 is an almost complete rewrite of the 18.04 dockerfile.

---

I can rebase my changes on top of #31, however, I saw there are several items on the TODO list.
Would it be better to do it after #31 get merged?

Thanks

---

@arthur-she If you mean the investigation of test failures on 20.04, those are follow-ups: the objective of #31 is to have a docker image with 20.04, but we won't yet be running the tests on it. I think #31 just needs a few minor changes. I don't know how soon @minosgalanakis can make them though — Minos, when do you think you'll have time to rework #31?

---

`outcomes.csv` is corrupted in a few places. It looks like a race condition where two concurrent threads were writing to the same file (either `outcomes.csv` itself, or one of the per-job files it's assembled from) and their writes got intermixed.

I've never seen this happen before. So either it's a very rare race condition that you were unlucky to be the first to discover (maybe due to different scheduling or load patterns on your Jenkins instance), or it's a race condition caused by your system having more concurrency somewhere.

I'll look a bit more closely to see if I can guess where the corruption happened.

---

Scratch that, the explanation is a lot simpler. Some jobs ran out of disk space at various times. For example [`all_u16-test_no_pem_no_fs
`](https://ci.staging.trustedfirmware.org/job/arthur-mbed-tls-nightly-tests/14/execution/node/977/log/) ran out during the build, [`all_u16-test_ref_configs`](https://ci.staging.trustedfirmware.org/job/arthur-mbed-tls-nightly-tests/14/execution/node/916/log/) ran out when it tried to update a state file (and then failed to update the Jenkins log as well?), and some ran out while writing to their outcome file, resulting in missing parts including line breaks.

---

Yeah, I rebuilt the tests and got the "no space left on device" error. Trying to increase the volume size of our AMI now.

---

After increasing the volume size, I got the [build](https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-nightly-tests/85/console) successfully. 
I also rebased my changes.
Please have a look.

Thanks